java-frontend/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -125,37 +126,41 @@ private ProgramState(ProgramState ps, PMap<SymbolicValue, Constraint> newConstra
   }
 
   ProgramState stackValue(SymbolicValue sv) {
-    Deque<SymbolicValue> newStack = new LinkedList<>(stack);
-    newStack.push(sv);
-    return new ProgramState(this, newStack);
+    return new ProgramState(this, stack.push(sv));
   }
 
   ProgramState clearStack() {
-    return unstackValue(stack.size()).state;
+    return stack.isEmpty() ? this : new ProgramState(this, PCollections.emptyStack());
   }
 
   public Pop unstackValue(int nbElements) {
-    if (nbElements == 0) {
-      return new Pop(this, Collections.<SymbolicValue>emptyList());
+    if (nbElements == 0 || stack.isEmpty()) {
+      return new Pop(this, Collections.emptyList());
     }
-    Preconditions.checkArgument(stack.size() >= nbElements, nbElements);
-    Deque<SymbolicValue> newStack = new LinkedList<>(stack);
+
+    // FIXME can be made more efficient by reusing sub collection of PStack instead of copying to the new list
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)